app/services/signal_execution_bridge.py

```
@@ -0,0 +1,102 @@
+        user: User,
+        execution_mode: str = "autonomous",
+    ) -> Dict[str, Any]:
+        if delivery_log.executed_at:
+            return {
+                "status": "already_executed",
+                "execution_reference": delivery_log.execution_reference,
+            }
+
+        event = delivery_log.event
+        if not event:
+            raise ValueError("Delivery log is not associated with a signal event")
```

Ruff (0.13.3)

Avoid specifying long messages outside the exception class

(TRY003)